src/components/Calendar/doubleCalendar/monthHeader.js

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useFormattedMonth } from './hooks';
+import StyledHeaderContainer from './styled/headerContainer';
+import StyledMonth from './styled/month';
+import YearSelect from './yearSelect';
+
+export default function MonthHeader(props) {
+    const { id, month, locale, currentYear, yearsRange, onYearChange } = props;
+    const formattedLabel = useFormattedMonth(month, locale);
```

Comment:
I think that in these situations the best option is to make useMemo or useCallback to the props and functions that are going to be passed in the parent components and the child components to do React.memo (). This ensures that the child components run only when necessary.